src/core/tests/Test_AptitudePackageManager.py

```diff
@@ -349,6 +373,117 @@ def test_update_image_default_patch_mode_raises_exception(self):
+        backup_package_manager_ubuntu_pro_client_install_or_update_pro = UbuntuProClient.UbuntuProClient.install_or_update_pro
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = self.mock_install_or_update_pro_raise_exception
+
+        obj = AptitudePackageManager.AptitudePackageManager(package_manager.env_layer, execution_config, package_manager.composite_logger, package_manager.telemetry_writer, package_manager.status_handler)
+
+        self.assertIsNotNone(obj)
+        self.assertIsNotNone(obj.ubuntu_pro_client)
+
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = backup_package_manager_ubuntu_pro_client_install_or_update_pro
+
+    def test_is_reboot_pending_pro_client_success(self):
+        version_mock = MockVersionResult()
+        version_mock.mock_import_uaclient_version_module('version', 'mock_version')
+        reboot_mock = MockRebootRequiredResult()
+        reboot_mock.mock_import_uaclient_reboot_required_module('reboot_required', 'mock_reboot_required_return_no')
+        runtime = RuntimeCompositor(ArgumentComposer().get_composed_arguments(), True, Constants.APT)
+        backup_AptitudePackageManager__pro_client_prereq_met = runtime.package_manager._AptitudePackageManager__pro_client_prereq_met
+        runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = True
+        self.assertFalse(runtime.package_manager.is_reboot_pending())
+
+        runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = backup_AptitudePackageManager__pro_client_prereq_met
+        version_mock.mock_unimport_uaclient_version_module()
+        reboot_mock.mock_unimport_uaclient_reboot_required_module()
+
+    def test_is_reboot_pending_test_mismatch(self):
+        version_mock = MockVersionResult()
+        version_mock.mock_import_uaclient_version_module('version', 'mock_version')
+        reboot_mock = MockRebootRequiredResult()
+        reboot_mock.mock_import_uaclient_reboot_required_module('reboot_required', 'mock_reboot_required_return_yes')
+        runtime = RuntimeCompositor(ArgumentComposer().get_composed_arguments(), True, Constants.APT)
+        backup_package_manager_is_reboot_pending = runtime.package_manager.ubuntu_pro_client.is_reboot_pending
+        runtime.package_manager.ubuntu_pro_client.is_reboot_pending = self.mock_is_reboot_pending_returns_True
+        runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = True
+
+        # test should return true as we fall back to Ubuntu Pro Client api`s result.
+        self.assertTrue(runtime.package_manager.is_reboot_pending())
+
+        runtime.package_manager.ubuntu_pro_client.is_reboot_pending = backup_package_manager_is_reboot_pending
+        version_mock.mock_unimport_uaclient_version_module()
+        reboot_mock.mock_unimport_uaclient_reboot_required_module()
+
+    def test_is_reboot_pending_test_raises_exception(self):
+        version_mock = MockVersionResult()
```

Comment:
There is no need of version_mock because setting runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = True

Also, there is no need of reboot_mock as setting runtime.package_manager.ubuntu_pro_client.is_reboot_pending = self.mock_is_reboot_pending_returns_False

Reply:
Done